src/iexamine.cpp

```diff
@@ -653,7 +653,7 @@ void iexamine::cardreader(player &p, const tripoint &examp)
 
 void iexamine::rubble(player &p, const tripoint &examp)
 {
-    bool has_digging_tool = p.has_quality( "DIG", 2 );
+    bool has_digging_tool = p.has_quality( quality_id( "DIG" ), 2 );
```

Comment:
------------------------------------
Mostly to encourage few explicit instantiating of it. Creating a temporary like here is fine if it's done only once (for that value) in a compilation unit. If it's done several times, one should use a constant, either function local like in `iuse::trap`, or global. This allows spellchecking by the compiler.

However, this line here is not the desired result. Ideally the condition would be `p.has( quality_id( "DIG" ), 2 )`, a function named `has` (or similar), overloaded for various id types. That gives a less redundant, but type safe interface: `if( has( some_trait_id ) && has( some_effect_id, bodypart ) && has( itype_id( "..." ) ) )`.

And it allows to *easily find all literal ids in the code.

@Coolthulhu I fixed it, I should probably use clang more often.